tests/table/test_between.py

```diff
@@ -0,0 +1,100 @@
+import datetime
+
+import pytest
+
+from daft import col
+from daft.table import MicroPartition
+
+
```

Comment:
Can you add some test cases to cover errors such as unsupported data types, e.g. https://github.com/Eventual-Inc/Daft/blob/main/tests/table/test_table_aggs.py#L229 and invalid array lengths e.g. https://github.com/Eventual-Inc/Daft/blob/main/tests/table/test_from_py.py#L565